src/nimony/deps.nim

```diff
@@ -359,15 +363,25 @@ proc generateFrontendMakefile(c: DepContext; commandLineArgs: string): string =
   result = c.config.nifcachePath / c.rootNode.files[0].modname & ".makefile"
   writeFile result, s
 
-proc generateCachedConfigFile(c: DepContext) =
-  let path = c.config.cachedConfigFile()
-  let configStr = c.config.getOptionsAsOneString() & " " & c.rootNode.files[0].nimFile
-  let needUpdate = if semos.fileExists(path) and not c.forceRebuild:
-                     configStr != readFile path
-                   else:
-                     true
-  if needUpdate:
-    writeFile path, configStr
+proc generateCachedConfigFile(c: DepContext; passC: string) =
+  block:
```

Comment:
Oh god no... Why does this need yet another tiny file?

Reply:
When only a passC command line option was changed, only the backend C compiler and linker need to be executed.
If passC was written to `cachedconfigfile.txt` like other command line options, nimsem, Hexer and NIFC are executed even if only the passC was changed and don't need to execute them.

Reply:
PassC changes are so rare that it doesn't matter, instead keep it simple.

Reply:
Created simpler version of this PR: https://github.com/nim-lang/nimony/pull/807